src/lib/registration/ui/service_selection_dialog.rb

```diff
@@ -0,0 +1,147 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "ui/dialog"
+require "registration/helpers"
+
+module Registration
+  module UI
+    # This class implements a SCC/SMT service selection dialog.
+    class ServiceSelectionDialog < ::UI::Dialog
+      # @return [Array<SlpServiceClass::Service] list of services to show
+      attr_reader :services
+
+      # Run dialog
+      #
+      # The return value will be:
+      # * A service in case one SMT server was selected
+      # * :scc symbol if default SCC was selected
+      # * :cancel symbol if the SCC was canceled (pressing the 'cancel' button)
+      #
+      # @example Select the default SCC service
+      #   Registration::UI::SelectionServiceDialog.run(services) #=> :scc
+      #
+      # @example Select some SMT service
+      #   Registration::UI::SelectionServiceDialog.run(services) #=> #<Yast::SlpServiceClass::Service...>
+      #
+      # @param services [Array<SlpServiceClass::Service] list of services to show
+      # @return [SlpServiceClass::Service,Symbol] selected service or symbol (:scc or :cancel)
+      #
+      # @see #run
+      def self.run(services)
+        new(services).run
+      end
+
+      # Constructor
+      #
+      # @param services [Array<SlpServiceClass::Service] list of services to show
+      def initialize(services = [])
+        super()
+
+        Yast.import "UI"
+        Yast.import "Label"
+        Yast.import "Report"
+
+        textdomain "registration"
+
+        @services = services
+      end
+
+      # Handler for the Ok button
+      #
+      # If no option was selected, a error message is shown.
+      def ok_handler
+        selected = Yast::UI.QueryWidget(Id(:services), :CurrentButton)
+        if !selected
+          Yast::Report.Error(_("No registration server selected."))
```

Comment:
Then I'd probably make it even more generic (like `"No service selected."` or something like that).